src/node/mempool_persist.cpp

```diff
@@ -58,15 +58,19 @@ bool LoadMempool(CTxMemPool& pool, const fs::path& load_path, Chainstate& active
     try {
         uint64_t version;
         file >> version;
-        std::vector<std::byte> xor_key;
         if (version == MEMPOOL_DUMP_VERSION_NO_XOR_KEY) {
-            // Leave XOR-key empty
+            file.SetXor(0);
         } else if (version == MEMPOOL_DUMP_VERSION) {
-            file >> xor_key;
+            std::vector<std::byte> xor_key_vector(8);
```

Comment:
Here and on line 185:
```suggestion
            std::vector<std::byte> xor_key_vector{8};
```

Reply:
Now I have to rename the PR :D